### ISSUE TITLE
Problem: fixWidth.jsx throws Invariant Violation in React v15

### ISSUE THREAD

**Issue by trevorbernard** [rrag/react-stockcharts#67]
> invariant.js:39 Uncaught Invariant Violation: traverseParentPath(...): Cannot traverse from and to the same ID, ``.

Code to reproduce

```javascript
import React from "react";
import d3 from "d3";

import ReStock from "react-stockcharts";

var {ChartCanvas, Chart} = ReStock;

var {CandlestickSeries} = ReStock.series;
var {XAxis, YAxis} = ReStock.axes;
var {fitWidth} = ReStock.helper;

class CandleStickChart extends React.Component {
  render() {
    const {type, width, data} = this.props;
    return (
      <ChartCanvas width={width} height={400}
                   margin={{left: 0, right: 0, top:25, bottom: 30}} type={type}
                   data={data}
                   seriesName='CandleStick'
                   xAccessor={d => d.date} xScale={d3.time.scale()}>

        <Chart id={1} yExtents={d => [d.high, d.low]}>
          <XAxis axisAt="bottom" orient="bottom" ticks={6} stroke="none" tickStroke="#9E9E9E"/>
          <YAxis axisAt="left" orient="right" ticks={10} stroke="none" tickStroke="#9E9E9E"/>
          <CandlestickSeries fill={d => d.close > d.open ? "#32c2b0" : "#EDB24E"}/>
        </Chart>
      </ChartCanvas>
    );
  }
}

CandleStickChart.propTypes = {
  data: React.PropTypes.array.isRequired,
  width: React.PropTypes.number.isRequired,
  type: React.PropTypes.oneOf(["svg", "hybrid"]).isRequired,
};

CandleStickChart.defaultProps = {
  type: "svg"
};
CandleStickChart = fitWidth(CandleStickChart);

export default CandleStickChart;
```

**Comment by trevorbernard** [comment]
Commenting out the fitWidth yields another problem when using react 15:

> Uncaught Invariant Violation: addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded (details: https://fb.me/react-refs-must-have-owner).

**Comment by rrag** [comment]
I used your code and I cannot recreate the problem with 0.4. Can you share a plunker please.

**Comment by rrag** [comment]
sorry, I did not read it properly. I have not tested this with react v15. Let me try that

**Comment by rrag** [comment]
http://plnkr.co/edit/BwvZLp0qLowypYpmuIEL?p=preview

cannot recreate. Used the dev build of react and react-stockcharts

**Comment by thisbejim** [comment]
I'm also having this issue on react-stockcharts: 0.4.0 and react: 15.0.1.

**Comment by trevorbernard** [comment]
When I get some free time, I'll make up a minimal test case. I use webpack and it's quite possible, it's an environmental issue.

**Comment by tommiley** [comment]
This happens when using npm, react-stockcharts is loading the older version of react and its conflicting with 15.0.1. Once react-stockcharts is updated to depend on 15.0.1 it should resolve the issue.

**Comment by rrag** [comment]
there is a [peerDependency](https://github.com/rrag/react-stockcharts/blob/master/package.json#L19-L21) of React 14, so I thought `npm install react-stockcharts` would fail.

Any case you are right, having multiple versions of react has been known to cause some problems. I will add 15 also to the dependency, peerDependency list for the next version.

**Comment by rrag** [comment]
fixed, published in 0.5
